app/views/play/CampaignView.coffee

```diff
@@ -613,7 +613,7 @@ module.exports = class CampaignView extends RootView
     false
 
   annotateLevels: (orderedLevels) ->
-    return if @course?
+    return if @CourseInstance?
```

Comment:
This seems like it would always return, since `@CourseInstance` is never defined.  Did you mean `@courseInstance`?

Does this change result in the correct user behavior?

Reply:
I did not notice that there is a typo, sorry for the mistake. Not executing annotate level is accidentally fixing the issue though. 
I fixed the typo and I am again able to sporadically reproduce this issue like last time. Sorry for the hasty check in again.  
Need to investigate this more.